SignInCheckIn/MinistryPlatform.Translation/Repositories/ChildSigninRepository.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.Collections.Generic;
+using MinistryPlatform.Translation.Models.DTO;
+using MinistryPlatform.Translation.Repositories.Interfaces;
+using System.Linq;
+using Crossroads.Utilities.Services.Interfaces;
+
+namespace MinistryPlatform.Translation.Repositories
+{
+    public class ChildSigninRepository : IChildSigninRepository
+    {
+        private readonly IApiUserRepository _apiUserRepository;
+        private readonly IMinistryPlatformRestRepository _ministryPlatformRestRepository;
+        private readonly IApplicationConfiguration _applicationConfiguration;
+
+        public ChildSigninRepository(IApiUserRepository apiUserRepository,
+            IMinistryPlatformRestRepository ministryPlatformRestRepository,
+            IApplicationConfiguration applicationConfiguration)
+        {
+            _apiUserRepository = apiUserRepository;
+            _ministryPlatformRestRepository = ministryPlatformRestRepository;
+            _applicationConfiguration = applicationConfiguration;
+        }
+
+        public List<MpParticipantDto> GetChildrenByPhoneNumber(string phoneNumber)
+        {
+            var householdId = GetHouseholdIdByPhoneNumber(phoneNumber);
+
+            if (householdId == -1) return new List<MpParticipantDto>();
+            var children = GetChildParticpantsByPrimaryHousehold(householdId);
+            GetChildParticpantsByOtherHousehold(householdId, children);
+            children = GetOnlyKidsClubChildren(children);
+
+            return children;
+        }
+
+        private int GetHouseholdIdByPhoneNumber(string phoneNumber)
+        {
+            var apiUserToken = _apiUserRepository.GetToken();
+
+            var columnList = new List<string>
+            {
+                "Contact_ID",
+                "Household_ID_Table.Household_ID",
+                "Household_Position_ID_Table.Household_Position_ID",
+                "Household_ID_Table.Home_Phone",
+                "Mobile_Phone",
+            };
+
+            var household = _ministryPlatformRestRepository.UsingAuthenticationToken(apiUserToken).
+                Search<MpContactDto>($"Household_Position_ID_Table.[Household_Position_ID] IN ({_applicationConfiguration.HouseHoldIdsThatCanCheckIn}) AND ([Mobile_Phone] = '{phoneNumber}' OR Household_ID_Table.[Home_Phone] = '{phoneNumber}')", columnList);
+
+            if (household == null || !household.Any())
```

Comment:
Refactor this to use a ternary operator or the new null coalesce, so we return at only one point